invenio_circulation/transitions/transitions.py

```diff
@@ -38,6 +38,35 @@ def _ensure_valid_loan_duration(loan):
         raise TransitionConstraintsViolation(msg=msg)
 
 
+def _ensure_valid_extension(loan):
+    """Validate end dates for a extended loan."""
+    get_extension_max_count = current_app.config['CIRCULATION_POLICIES'][
+        'extension']['max_count']
+    extension_max_count = get_extension_max_count(loan)
+
+    extension_count = loan.get('extension_count', 0)
+    extension_count += 1
+    if extension_max_count != -1 and extension_count > extension_max_count + 1:
```

Comment:
This implied logic of infinity is difficult to follow, and logically incorrect, because the `extension_count` here has to be strictly greater than the `max_count` increased by one, so its not the max. I would initialise the `extension_max_count` with a big number.